### PR TITLE
Make confluence_version a First Class Fact

### PR DESCRIPTION
**Pull request by spjmurray**
This stems from the fact that the confluence module unconditionally manages the
directory /etc/puppetlabs/facter/facts.d directory in the sense that if the code
gets evaluated first you run into the age old race condition of clashing with other
classes who wish to manage it.  This removes said management, moves the fact
to a ruby/puppet native which yields slightly prettier and less code.

The change to `confluence::service` was required to get CI to pass.  I'd properly
refactor this but it'd probably destroy users who already reference in hiera
`confluence::service::service_file_location` etc.

### CODE REVIEW

**Comment by spjmurray** [comment]
Also helps with #83 becoming a thing of the past!

**Comment by dustycodes** [comment]
Any idea when this will be merged?

**Comment by dhoppe** [comment]
@spjmurray Thx for your contribution.

File: lib/facter/confluence_version.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Facter.add(:confluence_version) do
+  setcode do
+    ps = Facter::Util::Resolution.exec('ps af')
```
Comment (dhoppe) [inline lib/facter/confluence_version.rb]: @spjmurray Are you sure this is the correct set of parameters?

```
root@ubuntu-14:~# ps af
  PID TTY      STAT   TIME COMMAND
 6617 pts/1    Ss     0:00 -bash
 9427 pts/1    S      0:00  \_ sudo su -
 9428 pts/1    S      0:00      \_ su -
 9429 pts/1    S      0:00          \_ -su
 9594 pts/1    R+     0:00              \_ ps af
 5492 tty1     Ss+    0:00 /sbin/getty -8 38400 tty1
  905 tty6     Ss+    0:00 /sbin/getty -8 38400 tty6
  903 tty3     Ss+    0:00 /sbin/getty -8 38400 tty3
  902 tty2     Ss+    0:00 /sbin/getty -8 38400 tty2
  898 tty5     Ss+    0:00 /sbin/getty -8 38400 tty5
  896 tty4     Ss+    0:00 /sbin/getty -8 38400 tty4
```

```
root@ubuntu-14:~# ps ax | grep atlassian
 5550 ?        Sl     0:58 /usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java -Djava.util.logging.config.file=/opt/confluence/atlassian-confluence-5.7.1/conf/logging.properties -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager -Xms256m -Xmx1024m -XX:MaxPermSize=256m -Djava.awt.headless=true -Djava.endorsed.dirs=/opt/confluence/atlassian-confluence-5.7.1/endorsed -classpath /opt/confluence/atlassian-confluence-5.7.1/bin/bootstrap.jar:/opt/confluence/atlassian-confluence-5.7.1/bin/tomcat-juli.jar -Dcatalina.base=/opt/confluence/atlassian-confluence-5.7.1 -Dcatalina.home=/opt/confluence/atlassian-confluence-5.7.1 -Djava.io.tmpdir=/opt/confluence/atlassian-confluence-5.7.1/temp org.apache.catalina.startup.Bootstrap start
```

**Comment by dhoppe** [comment]
@spjmurray I think you also need to make sure that the old fact `confluence_facts.sh` does not exist anymore.

**Pull request merged**